floodlight/core/events.py

```diff
@@ -61,3 +74,155 @@ def custom(self):
+
+        Returns
+        -------
+        check: bool
+            True if the checks for all protected columns pass and False otherwise
+        """
+        check = True
+        for col in self.protected:
+            check *= self.definition_check(col, protected_columns)
+
+        return check
+
+    def definition_check(self, col: str, definitions: Dict[str, Dict]) -> bool:
+        """Perform the checks for value range for a single column of the inner event
+        DataFrame using the specifications from floodlight.core.definitions.
+
+        Parameters
+        ----------
+        col: str
+            Column of the inner event DataFrame to be checked
+        definitions: Dict
+            Dictionary for each column name to be checked. Each entry is a Dictionary
+            containing definitions for allowed value ranges of the respective
+            column. Information about value_range is given as a List of the form
+            [min_value, max_value, extra_value] where the extra_value indicates an
+            additionally allowed value that does not fit within the defined value range.
+
+        Returns
+        -------
+        check: bool
+            True if the checks for value range pass and False otherwise
+        """
+        check = True
+
+        # check if value range is defined
+        if definitions[col]["value_range"] is None:
+            return check
+
+        # skip extra allowed values
+        check_col = self.events[col]
+        if len(definitions[col]["value_range"]) == 3:
+            if definitions[col]["value_range"][2] is None:
+                check_col = check_col.dropna()
+            else:
+                check_col = check_col[check_col != definitions[col]["value_range"][2]]
+
+        # check value range for remaining values
+        if not (definitions[col]["value_range"][0] <= check_col).all():
+            check = False
+            warnings.warn(
+                f"Events column {col} has at least one value smaller than "
+                f"{definitions[col]['value_range'][0]}!"
+            )
+        if not (check_col <= definitions[col]["value_range"][1]).all():
+            check = False
+            warnings.warn(
+                f"Events column {col} has at least one value larger than "
+                f"{definitions[col]['value_range'][1]}!"
+            )
+
+        return check
+
+    def add_frameclock(self, framerate: int):
+        """Add the column "frameclock", computed as the rounded multiplication of
+        gameclock and framerate, to the inner events DataFrame.
+
+        Parameters
+        ----------
+        framerate: int
+            Temporal resolution of data in frames per second/Hertz.
+        """
+        frameclock = np.full((len(self.events)), -1, dtype=int)
+        frameclock[:] = np.round(self.events["gameclock"].values * framerate)
+        self.events["frameclock"] = frameclock
+
+    def find(self, col: str, value: Any) -> pd.Series:
```

Comment:
Maybe this functionality should be a little more advanced, e.g. by including multiple search parameters. 

Right now, If I'm not mistaken, you could do the same with events[events[col]==value], which returns the whole (filtered) df. That's not too much of a difference to events.find(col, value). I also noticed you return a Series, is there any reason for this?

Reply:
Ok, I integrated the possibility to include mutliple checks by allowing input arguments 
[(column1, condition1), (column2, condition2), ...].

However, it may now be confusing that self.find(column1, condition1) throws an error.  Do you think this is a problem or is it most pythonic to just implicitly enforce self.find([(column1, condition1)])?